blocks/algorithms/__init__.py

```diff
@@ -293,23 +297,26 @@ class CompositeRule(StepRule):
     def __init__(self, components):
         self.components = components
 
-    def compute_steps(self, gradients):
-        result = gradients
+    def compute_steps(self, previous_steps):
+        steps = previous_steps
         updates = []
         for rule in self.components:
-            result, more_updates = rule.compute_steps(result)
+            steps, more_updates = rule.compute_steps(steps)
             updates += more_updates
-        return result, updates
+        return steps, updates
+
 
+class Scale(StepRule):
+    """A step in the direction proportional to the previous step.
 
-class SteepestDescent(StepRule):
-    """A step in the direction proportional to the gradient.
+    If the previous step is the gradient itself, this step rule is
+    equivalent to steepest ascent.
```

Comment:
Why is it ascent, not descent? Yes, what it returns points in the same direction as the gradient, but what it returns shall be subtracted, which effectively means that it implement steepest descent.